src/renderer/components/+apps-helm-charts/helm-charts.tsx

```diff
@@ -72,9 +72,6 @@ export class HelmCharts extends Component<Props> {
             (chart: HelmChart) => chart.getAppVersion(),
             (chart: HelmChart) => chart.getKeywords(),
           ]}
-          filterItems={[
-            (items: HelmChart[]) => items.filter(item => !item.deprecated)
-          ]}
```

Comment:
Why was this not working as intended? Was it filtering too late?

Reply:
`filterItems` is (being) deprecated from `ItemListLayout` anyway...?

Reply:
Ah I see makes sense

Reply:
Yes, `filterItems` would get deprecated in some way, but that's not the only reason. Before this PR we had filters in both places - here in UI in `helm-charts.tsx` and inside `helm-service.ts`. So, keeping it in both places doesn't make any sense and I chose to leave it in "backend" with some refactorings (where my new class was introduced). My decision was dictated by a feeling that react component shouldn't do such actions (maybe in rare cases only).

Reply:
Sounds good